src/main/java/FindCommand.java

```diff
@@ -0,0 +1,33 @@
+import java.util.ArrayList;
+
+public class FindCommand extends Command {
+    protected String keyWord;
+
+    public FindCommand(String keyWord) {
+        this.keyWord = keyWord;
+    }
+
+    @Override
+    public boolean isExit() {
+        return false;
+    }
+
+    @Override
+    public void execute (TaskList tasks, Ui ui, Storage storage) {
+        ArrayList<String> matchList = new ArrayList<String>();
+        int index = 0;
+
+        for (Task task : tasks.tasks) {
+            String line = task.toString();
+            if (line.contains(keyWord)) {
+                matchList.add(line);
+            }
+        }
+
+        System.out.println("Here are the matching tasks in your list:");
+
+        for (String line : matchList) {
+            System.out.println((index + 1) + "." + line);
```

Comment:
It wasn't explicitly specified in the instructions whether to assign new indexes to the matching tasks (numbering them from 1 again), but it does imply that in the instructions. 

As for me, I kept the original indexes of the task so that the user can more easily refer to their index if they want to issue a Done or Delete command. 

The instructions are ambiguous and this is something I will clarify with Prof and my tutor, so just a heads up to keep a lookout for clarifications regarding this!